### PR TITLE
Chat implementation

### PR DESCRIPTION
**Pull request by mihai9112**

### CODE REVIEW

File: ios/Flutter/flutter_export_environment.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (mihai9112) [inline ios/Flutter/flutter_export_environment.sh]: @ujwalbasnet1 please remove this file

File: lib/src/chat/chat_page.dart [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+  @override
+  _ChatPageState createState() => _ChatPageState();
+}
+
+class _ChatPageState extends State<ChatPage> {
+  String UserID = 'OMA4VjyncrWeIlGIrGtVwLpLe3D3';
+
+  @override
+  void initState() {
+    // ujwal - sender  - OMA4VjyncrWeIlGIrGtVwLpLe3D3
```
Comment (mihai9112) [inline lib/src/chat/chat_page.dart]: @ujwalbasnet1 remove this

File: lib/src/chat/chat_bloc/chat_entity.dart [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      userId: snap.data['userId'] as String,
+      imgURL: snap.data['imgURL'] as String ??
+          'https://img.webmd.com/dtmcms/live/webmd/consumer_assets/site_images/article_thumbnails/reference_guide/baby_development_your_3_month_old_ref_guide/650x350_baby_development_your_3_month_old_ref_guide.jpg',
+      unreadMessageCount: snap.data['unreadMessageCount'] as int ?? 12,
+      latestMessage: snap.data['latestMessage'] as String ?? 'Hello',
+      isActive: snap.data['isActive'] as bool ?? false,
+    );
+  }
+
+//  Map<String, Object> toDocument() {
```
Comment (mihai9112) [inline lib/src/chat/chat_bloc/chat_entity.dart]: @ujwalbasnet1  remove this

File: lib/src/chat/chat_page.dart [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+import 'message_model.dart';
+import 'message_page.dart';
+
+class ChatPage extends StatefulWidget {
+  @override
+  _ChatPageState createState() => _ChatPageState();
+}
+
+class _ChatPageState extends State<ChatPage> {
+  String UserID = 'OMA4VjyncrWeIlGIrGtVwLpLe3D3';
```
Comment (mihai9112) [inline lib/src/chat/chat_page.dart]: @ujwalbasnet1 does this needs to be hardcoded?

File: lib/src/chat/chat_page.dart [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+//List<ChatItemModel> getMessageItemList() {
```
Comment (mihai9112) [inline lib/src/chat/chat_page.dart]: @ujwalbasnet1 please remove

File: lib/src/chat/chat_repository.dart [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+  String toString() {
+    // TODO: implement toString
+    return 'firstName: $firstName, lastName: $lastName, user_id: $userId';
+  }
+}
+
+class FireBaseChatRepository {
+  @override
+  Future<List<User>> getChattedUsers() async {
+//    var temp = Firestore.instance
```
Comment (mihai9112) [inline lib/src/chat/chat_repository.dart]: @ujwalbasnet1 please remove

**Review by mihai9112 — COMMENTED** [review]
I cannot build the project locally so I cannot review it. Please fix it before I'm able to review the code:
1). I cannot find any package `equatable` in the `pubspec.yaml`, do you use this package? How do you import it on your machine?
2). Class `ChatRepository` does not exist. There is a class called `FireBaseChatRepository` but it doesn't have the correct methods implemented.

Please change as I'm unable to continue

Also, please merge master in your branch. 

@ujwalbasnet1

File: lib/src/chat/chat_page.dart (hunk 4 above)
Comment (ujwalbasnet1) [reply lib/src/chat/chat_page.dart]: no this will be stored in sharedPreferences.

**Pull request closed without merging**